hipsrc/attention/attention_kernels.hip

```diff
@@ -0,0 +1,524 @@
+/*
+ * Adapted from https://github.com/NVIDIA/FasterTransformer/blob/release/v5.3_tag/src/fastertransformer/kernels/decoder_masked_multihead_attention/decoder_masked_multihead_attention_template.hpp
+ * Copyright (c) 2023, The vLLM team.
+ * Copyright (c) 2020-2023, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <hip/hip_runtime.h>
+#include <torch/extension.h>
+#include <ATen/hip/HIPContext.h>
+
+#include "attention_dtypes.h"
+#include "attention_utils.hip"
+
+#include <algorithm>
+
+#define WARP_SIZE 32
```

Comment:
Warp size can be changed to 64 in case of rocm, reference: accelerated computing with HIP, the book, chapter 8.4, page 116

Reply:
Great spot, changed warp size to 64